src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double lowerBase;
+    private double upperBase;
+    private double side;
+
+    public IsoscelesTrapezoid(Color color, double lowerBase, double upperBase, double side) {
+        super(color);
+        this.lowerBase = lowerBase;
+        this.upperBase = upperBase;
+        this.side = side;
+    }
+
+    @Override
+    public double calculateArea() {
```

Comment:
The formula for the area calculation is complex and might be hard to read. Consider breaking it down into smaller parts and using descriptive variable names for each part.